### PR TITLE
[EOC-58] Keyoboard support for dialogs (ESCAPE and Click)

### PR DESCRIPTION
**Pull request by adamklepacz**

### CODE REVIEW

File: client/src/modules/shopping-list/index.jsx [hunk 1]
```diff
@@ -53,12 +53,36 @@ class ShoppingList extends Component {
     fetchItemsFromGivenList(id);
   };
 
+  handleClick = event => {
```
Comment (alemikolo) [inline client/src/modules/shopping-list/index.jsx]: we have a mess with handlers names... here is handleClick, and escapeHandler (below). In toolbar.jsx is clickListener and escapeListener...

File: client/src/modules/shopping-list/index.jsx [hunk 2]
```diff
@@ -53,12 +53,36 @@ class ShoppingList extends Component {
...
+    const { className } = event.target;
+
+    className.length > 0 && className.includes('overlay')
+      ? this.setState({ showDialogBox: false, showUpdateForm: false })
+      : null;
+  };
+
+  escapeHandler = () => {
+    this.setState({ showDialogBox: false, showUpdateForm: false });
```
Comment (alemikolo) [inline client/src/modules/shopping-list/index.jsx]: event handlers naming - see above

File: client/src/common/components/DialogBox.jsx [hunk 3]
```diff
@@ -1,17 +1,52 @@
...
+    const { code } = event;
+    const { onCancel } = this.props;
+
+    if (code === 'Escape') {
+      onCancel && onCancel();
+    }
+  };
+
+  render() {
+    const { onCancel, onConfirm, message } = this.props;
```
Comment (alemikolo) [inline client/src/common/components/DialogBox.jsx]: alphabetic order?

File: client/src/modules/shopping-list/index.jsx [hunk 4]
```diff
@@ -146,16 +120,14 @@ class ShoppingList extends Component {
...
-            <DialogBox
-              onCancel={this.hideDialogBox}
-              onConfirm={this.deleteListHandler(listId)}
-              message="Do you really want to delete the list?"
-            />
-          </ModalBox>
+          <DialogBox
+            onCancel={this.hideDialogBox}
+            onConfirm={this.deleteListHandler(listId)}
+            message="Do you really want to delete the list?"
```
Comment (alemikolo) [inline client/src/modules/shopping-list/index.jsx]: alphabetic order

File: client/src/modules/shopping-list/index.jsx [hunk 5]
```diff
@@ -146,16 +120,14 @@ class ShoppingList extends Component {
...
+          />
         )}
         {showUpdateForm && (
-          <ModalBox onClose={this.hideUpdateForm}>
+          <ModalBox onCancel={this.hideUpdateForm}>
             <CreationForm
               type="modal"
               label="Edit list"
```
Comment (alemikolo) [inline client/src/modules/shopping-list/index.jsx]: alphabetic order

File: client/src/modules/shopping-list/index.jsx [hunk 6]
```diff
@@ -120,20 +120,18 @@ class ShoppingList extends Component {
...
+          <DialogBox
+            message="Do you really want to delete the list?"
+            onCancel={this.hideDialogBox}
+            onConfirm={this.deleteListHandler(listId)}
+          />
         )}
-        {showUpdateFrom && (
-          <ModalBox onClose={this.hideUpdateForm}>
+        {showUpdateForm && (
+          <ModalBox onCancel={this.hideUpdateForm}>
```
Comment (marekrozmus) [inline client/src/modules/shopping-list/index.jsx]: this modal box would need to be refactored so that it is used inside components not outside of them

**Pull request merged**